pkg/controller/performanceprofile/components/tuned/tuned.go

```diff
@@ -55,19 +57,45 @@ func NewNodePerformance(assetsDir string, profile *performancev1.PerformanceProf
 	}
 
 	if profile.Spec.HugePages != nil {
+		var defaultHugepageSize performancev1.HugePageSize
 		if profile.Spec.HugePages.DefaultHugePagesSize != nil {
-			templateArgs[templateDefaultHugepagesSize] = string(*profile.Spec.HugePages.DefaultHugePagesSize)
+			defaultHugepageSize = *profile.Spec.HugePages.DefaultHugePagesSize
+			templateArgs[templateDefaultHugepagesSize] = string(defaultHugepageSize)
 		}
 
-		hugepages := []string{}
+		var is2MHugepagesRequested *bool
```

Comment:
just curious,  may be it's better to add dummy arguments for any hugepages sizes (not only 2M) when numa node is selected?

I don't see any issues with adding dummies for 1G as well, as for me in this case the code logic will be more readable/clearer.

Reply:
I prefer to avoid adding hack values when we do not really need it, I also have hope that in the near future we will have some work under the kernel that will preserve folders and files for all hugepages always